src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -20,42 +20,56 @@
 
 namespace OpenTelemetry.Instrumentation.Process;
 
-internal class ProcessMetrics
+internal sealed class ProcessMetrics
 {
     internal static readonly AssemblyName AssemblyName = typeof(ProcessMetrics).Assembly.GetName();
-    internal static readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
-    private static readonly Diagnostics.Process CurrentProcess = Diagnostics.Process.GetCurrentProcess();
+    internal readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
 
-    static ProcessMetrics()
+    private readonly Diagnostics.Process currentProcess = Diagnostics.Process.GetCurrentProcess();
+    private double? memoryUsage;
+    private double? virtualMemoryUsage;
+
+    public ProcessMetrics(ProcessInstrumentationOptions options)
     {
         // TODO: change to ObservableUpDownCounter
-        MeterInstance.CreateObservableGauge(
+        this.MeterInstance.CreateObservableGauge(
             "process.memory.usage",
             () =>
             {
-                CurrentProcess.Refresh();
-                return CurrentProcess.WorkingSet64;
+                if (!this.memoryUsage.HasValue)
```

Comment:
For these callbacks to be thread-safe they need to be inside a lock. For example:
```C#
() =>
{
    lock(this)
    {
        if (!this.memoryUsage.HasValue)
        {
            this.Snapshot();
        }
        var value = this.memoryUsage.Value;
        this.memoryUsage = null;
        return value;
    }
}
```